syntax = "proto3";

package crypto_condor;

// A single x25519 test vector.
message X25519Test{
	// The test ID, unique in its set of vectors.
	int32 id = 1;
	// The type of test. One of: valid, invalid, acceptable.
	string type = 2;
	// A comment on the test.
	string comment = 3;
	// Flags that categorize this test.
	repeated string flags = 4;

	// The raw "own" private key.
	bytes sk = 5;
	// The raw "peer" public key.
	bytes pk = 6;
	// The resulting shared secret.
	bytes shared = 7;
}

// A set of x25519 test vectors.
message X25519Vectors{
	// The source of the test vectors.
	string source = 1;
	// Description of the source.
	string source_desc = 2;
	// The URL of the source.
	string source_url = 3;
	// Whether these are compliance test vectors or not.
	bool compliance = 4;
	// A dictionary of test flags and their description.
    map<string, string> notes = 5;
	// The test vectors.
	repeated X25519Test tests = 6;
}
